packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2372,35 +2364,30 @@ abstract class RenderObject with DiagnosticableTreeMixin implements HitTestTarge
     markParentNeedsLayout();
   }
 
-  /// Set [_relayoutBoundary] to null throughout this render object's subtree,
+  /// Set [_isRelayoutBoundary] to null throughout this render object's subtree,
   /// stopping at relayout boundaries.
   // This is a static method to reduce closure allocation with visitChildren.
   static void _cleanChildRelayoutBoundary(RenderObject child) {
-    if (child._relayoutBoundary != child) {
+    if (child._isRelayoutBoundary != true) {
       child.visitChildren(_cleanChildRelayoutBoundary);
-      child._relayoutBoundary = null;
+      child._isRelayoutBoundary = null;
     }
   }
 
   // This is a static method to reduce closure allocation with visitChildren.
   static void _propagateRelayoutBoundaryToChild(RenderObject child) {
```

Comment:
maybe rename this to indicate that it will mark the child to NOT be a relayout boundary.